shell/common/snapshot_controller_impeller.cc

```diff
@@ -21,37 +24,84 @@ sk_sp<DlImage> DoMakeRasterSnapshot(
     SkISize size,
     const std::shared_ptr<impeller::AiksContext>& context) {
   TRACE_EVENT0("flutter", __FUNCTION__);
+  if (!context) {
+    return nullptr;
+  }
+  // Determine render target size.
+  auto max_size = context->GetContext()
+                      ->GetResourceAllocator()
+                      ->GetMaxTextureSizeSupported();
+  double scale_factor_x =
+      static_cast<double>(max_size.width) / static_cast<double>(size.width());
+  double scale_factor_y =
+      static_cast<double>(max_size.height) / static_cast<double>(size.height());
+  double scale_factor = std::min(1.0, std::min(scale_factor_x, scale_factor_y));
```

Comment:
IIRC, std::min has an overload that takes an initializer list. So you should be able do do just `std::min({a, b, c})` instead of `std::min(a, std::min(b, c))`.

Reply:
Done